sklearn_numba_dpex/common/random.py

```diff
@@ -0,0 +1,307 @@
+import warnings
+import random
+from functools import lru_cache
+
+from numba import float32, float64, uint32, int64, uint64
+import numpy as np
+
+import dpctl
+import dpctl.tensor as dpt
+import numba_dpex as dpex
+
+# This code is largely inspired from the numba.cuda.random module and the
+# numba/cuda/random.py where it's defined (v<0.57), and by the implementation of the
+# same algorithm in the package `randomgen`.
+
+# numba.cuda.random :https://github.com/numba/numba/blob/0.56.3/numba/cuda/random.py
+# randomgen: https://github.com/bashtage/randomgen/blob/v1.23.1/randomgen/xoroshiro128.pyx
+
+# NB1: we implement xoroshiro128++ rather than just xoroshiro128+, which is preferred.
+# Reference resource about PRNG: https://prng.di.unimi.it/
+
+# NB2: original numba.cuda.random code also includes functions for generating normally
+# distributed floats but we don't include it here as long as it's not needed.
+
+zero_idx = int64(0)
+one_idx = int64(1)
+
+
+def get_random_raw(states):
+    result = dpt.empty(sh=(1,), dtype=np.uint64, device=states.device)
+    make_random_raw_kernel()(states, result)
+    return result
+
+
+@lru_cache
+def make_random_raw_kernel():
+    @dpex.kernel
+    def _get_random_raw_kernel(states, result):
+        result[zero_idx] = _xoroshiro128pp_next(states, zero_idx)
+
+    return _get_random_raw_kernel[1, 1]
```

Comment:
Can you add docstrings to specify the kind of pseudo-random value this is generating (I assume float32 in [0, 1).)?